beams/tree_config.py

```diff
@@ -0,0 +1,298 @@
+
+
+@as_tagged_union
+@dataclass
+class BaseItem:
+    name: str = ''
+    description: str = ''
+
+    def get_tree(self) -> Behaviour:
+        """Get the tree node that this dataclass represents"""
+        raise NotImplementedError
+
+
+@dataclass
+class ExternalItem(BaseItem):
+    path: str = ''
+
+    def get_tree(self) -> Behaviour:
+        # grab file
+        # de-serialize tree, return it
+        raise NotImplementedError
+
+
+class ParallelMode(Enum):
+    """Simple enum mimicing the ``py_trees.common.ParallelPolicy`` options"""
+    Base = "Base"
+    SuccessOnAll = "SuccesOnAll"
+    SuccessOnONe = "SuccessOnOne"
+    SuccessOnSelected = "SuccessOnSelected"
+
+
+@dataclass
+class ParallelItem(BaseItem):
+    policy: ParallelMode = ParallelMode.Base
+    children: Optional[List[BaseItem]] = field(default_factory=list)
+
+    def get_tree(self) -> Parallel:
+        children = []
+        for child in self.children:
+            children.append(child.get_tree())
+
+        node = Parallel(
+            name=self.name,
+            policy=getattr(ParallelPolicy, self.policy.value),
+            children=children
+        )
+
+        return node
+
+
+@dataclass
+class SelectorItem(BaseItem):
+    """aka fallback node"""
+    memory: bool = False
+    children: Optional[List[BaseItem]] = field(default_factory=list)
+
+    def get_tree(self) -> Selector:
+        children = []
+        for child in self.children:
+            children.append(child.get_tree())
+
+        node = Selector(
+            name=self.name,
+            memory=self.memory,
+            children=children
+        )
+        return node
+
+
+@dataclass
+class SequenceItem(BaseItem):
+    memory: bool = False
+    children: Optional[List[BaseItem]] = field(default_factory=list)
+
+    def get_tree(self) -> Sequence:
+        children = []
+        for child in self.children:
+            children.append(child.get_tree())
+
+        node = Sequence(
+            name=self.name,
+            memory=self.memory,
+            children=children
+        )
+
+        return node
+
+
+# Custom LCLS-built Behaviors (idioms)
+class ConditionOperator(Enum):
+    equal = 'eq'
+    not_equal = 'ne'
+    less = 'lt'
+    greater = 'gt'
+    less_equal = 'le'
+    greater_equal = 'ge'
+
+
+@dataclass
+class ConditionItem(BaseItem):
+    pv: str = ''
+    value: Any = 1
+    operator: ConditionOperator = ConditionOperator.equal
+
+    def get_tree(self) -> ConditionNode:
+        cond_func = self.get_condition_function()
+        return ConditionNode(self.name, cond_func)
+
+    def get_condition_function(self) -> Callable[[], bool]:
+        op = getattr(operator, self.operator.value)
+
+        def cond_func():
+            val = caget(self.pv)
+            if val is None:
+                return False
+
+            return op(val, self.value)
+
+        return cond_func
+
+
+@as_tagged_union
+@dataclass
+class ActionItem(BaseItem):
+    loop_freq: float = 1.0
```

Comment:
crazy pedantic, I may have even suggested the original name. 
```suggestion
    loop_sleep_sec: float = 1.0
```

Reply:
Critically this is the period, not the frequency 🤦

Reply:
True too, but as other things may also take up clock cycles in the action I think indicating it as a sleep could be helpful. Semantics, whatever you pick is good.